modules/apps/commerce/commerce-client-extension-web/src/main/java/com/liferay/commerce/client/extension/web/internal/type/deployer/CommerceCETDeployerImpl.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.client.extension.web.internal.type.deployer;
+
+import com.liferay.client.extension.type.CET;
+import com.liferay.client.extension.type.CustomCheckoutStepCET;
+import com.liferay.client.extension.type.deployer.CommerceCETDeployer;
+import com.liferay.commerce.client.extension.web.internal.util.CustomCheckoutStep;
+import com.liferay.commerce.util.CommerceCheckoutStep;
+import com.liferay.frontend.taglib.servlet.taglib.util.JSPRenderer;
+import com.liferay.portal.catapult.PortalCatapult;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.service.UserService;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletContext;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(service = CommerceCETDeployer.class)
+public class CommerceCETDeployerImpl implements CommerceCETDeployer {
```

Comment:
Looks like a generic commerce custom element deployer, but it is in fact specific for the custom checkout step type.
Shouldn't we better naming this then?
And if in the future we add other types of CET we can reuse the module and just create new deployers ecc

Reply:
it is true that now it only holds the checkout CX, but if we'd need to implement new CX types for Commerce we just need to add the cases in the deploy method, like it is done in the original CETDeployer